mautrix_signal/portal.py

```diff
@@ -725,6 +725,43 @@ async def handle_matrix_leave(self, user: u.User) -> None:
                 await self.signal.leave_group(user.username, self.chat_id)
             # TODO cleanup if empty
 
+    async def kick_matrix(self, user: u.User | p.Puppet, source: u.User) -> None:
+        try:
+            await self.signal.update_group(
+                source.username, self.chat_id, remove_members=[user.address]
+            )
+        except Exception as e:
+            self.log.exception(f"Failed to kick Signal user: {e}")
+            await self.main_intent.invite_user(
+                self.mxid, user.mxid, check_cache=True, reason=f"Failed to kick Signal user: {e}"
+            )
+            await user.intent_for(self).ensure_joined(self.mxid)
```

Comment:
Hmm, would update_group throw an error if the user is not present in the group? (i.e. will kicking a ghost who isn't in the signal chat cause the ghost to rejoin, even though they aren't in the chat?)

Reply:
good point. Same goes for ban and unban. We should probably check the signal group in all three cases.